psi4/src/psi4/libmints/matrix.h

```diff
@@ -863,16 +863,8 @@ class PSI_API Matrix : public std::enable_shared_from_this<Matrix> {
 
     /// @{
     /// Diagonalizes this, eigvectors and eigvalues must be created by caller.  Only for symmetric matrices.
-    void diagonalize(Matrix* eigvectors, Vector* eigvalues, diagonalize_order nMatz = ascending);
+    void diagonalize(Matrix& eigvectors, Vector& eigvalues, diagonalize_order nMatz = ascending);
     void diagonalize(SharedMatrix& eigvectors, std::shared_ptr<Vector>& eigvalues, diagonalize_order nMatz = ascending);
-    void diagonalize(SharedMatrix& eigvectors, Vector& eigvalues, diagonalize_order nMatz = ascending);
-    /// @}
-
-    /// @{
-    /// Diagonalizes this, applying supplied metric, eigvectors and eigvalues must be created by caller.  Only for
-    /// symmetric matrices.
-    void diagonalize(SharedMatrix& metric, SharedMatrix& eigvectors, std::shared_ptr<Vector>& eigvalues,
```

Comment:
Am I reading this and the sq_rsp PR right that the diagonalize fn with `metric` parameter are going away?

Reply:
Yes? This PR is removing the "diagonalize with metric" fn from `Matrix` as it is never called by anything, and the later PR is not adding it back either, for the same reason.

Reply:
Ok, I'm not enthusiastic about removing the option. But 10y is a good span of opportunity for a use to be found. :-)